miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/storage/database/MiniAppSecureDatabase.kt

```diff
@@ -0,0 +1,251 @@
+package com.rakuten.tech.mobile.miniapp.storage.database
+
+import android.annotation.SuppressLint
+import android.content.ContentValues
+import android.content.Context
+import android.database.sqlite.SQLiteDatabase
+import android.database.sqlite.SQLiteException
+import androidx.annotation.NonNull
+import androidx.sqlite.db.SupportSQLiteDatabase
+import java.io.IOException
+import java.sql.SQLException
+
+private const val DB_HEADER_SIZE = 100
+private const val PAGE_SIZE_MULTIPLIER = 3
+private const val TABLE_NAME = "MiniAppCache"
+private const val FIRST_COLUMN_NAME = "first"
+private const val SECOND_COLUMN_NAME = "second"
+
+private const val GET_ALL_ITEMS_QUERY = "select * from $TABLE_NAME"
+private const val DROP_TABLE_QUERY = "DROP TABLE IF EXISTS $TABLE_NAME"
+private const val GET_ITEM_QUERY_PREFIX = "select * from $TABLE_NAME where $FIRST_COLUMN_NAME="
+private const val CREATE_TABLE_QUERY = "create table if not exists $TABLE_NAME (" +
+        "$FIRST_COLUMN_NAME text primary key, $SECOND_COLUMN_NAME text)"
+
+internal const val DATABASE_DOES_NOT_EXIST_ERROR = "Database does not exist."
+internal const val MAX_DB_SPACE_LIMIT_REACHED_ERROR =
+    "Can't insert new items. Database reached to max space limit."
+
+/**
+ * Concrete MiniApp Database Implementation.
+ */
+internal class MiniAppSecureDatabase(
+    @NonNull private var context: Context,
+    @NonNull dbName: String, // MiniAppId will be the dbName
+    @NonNull dbVersion: Int,
+    @NonNull private var maxDatabaseSize: Long
+) : MiniAppSecureDatabaseImpl(context, dbName, dbVersion) {
+
+    private lateinit var database: SupportSQLiteDatabase
+
+    private fun getDatabasePageSize(): Long {
+        return database.pageSize
+    }
+
+    override fun onCreateDatabase(db: SupportSQLiteDatabase) {
+        try {
+            db.execSQL(CREATE_TABLE_QUERY)
+            db.maximumSize = maxDatabaseSize
+        } catch (e: SQLException) {
+            // Ignoring.
```

Comment:
and here :P Basically all the empty catches

Reply:
@Climbatize Actually this one is needed here since `execSQL` throws `SQLException` So catching it in case to avoid an app crash. However I'll throw that exception from here and will return an `onFailed(FatalIOException)` in the dispatcher from the catch.